charts/kube-prometheus-stack/values.yaml

```diff
@@ -1414,10 +1414,25 @@ kubelet:
       - sourceLabels: [__name__]
         action: drop
         regex: 'container_(file_descriptors|tasks_state|threads_max)'
+      # Drop container_memory_failures_total{scope="hierarchy"} metrics,
+      # we only need the container scope.
+      - sourceLabels: [__name__, scope]
+        action: drop
+        regex: 'container_memory_failures_total;hierarchy'
+      # Drop container_network_... metrics that match various interfaces that
+      # correspond to CNI and similar interfaces. This avoids capturing network
+      # metrics for host network containers.
+      - sourceLabels: [__name__, interface]
+        action: drop
+        regex: 'container_network_.*;(cali|cilium|cni|lxc|nodelocaldns|tunl).*'
       # Drop container spec metrics that overlap with kube-state-metrics.
       - sourceLabels: [__name__]
         action: drop
         regex: 'container_spec.*'
+      # Drop POD / non-container metrics where we only care about container level metrics.
```

Comment:
I'm not sure about this. There specific, but unknown condition where is the cadvisor does report container metrics without container metric.

There are various issues like https://github.com/prometheus-community/helm-charts/issues/4972 that indicate that situation. Ref: https://github.com/kubernetes/kubernetes/issues/89903

I found a old comment from you https://github.com/k3s-io/k3s/issues/5891#issuecomment-1855885837 that indicates that label without container should be dropped always. Not sure about k3s, but some users report this behavior on AWS EKS as well.

In case, we agree to not support container metrics without labels, it should be more visible documented.

Reply:
Sure, I can remove this one. Having these extra metrics makes `sum without (container)` more complicated, but people have been living with this for a while.

Reply:
I would post-pone this clarify it in a distinct PR/Issue. Grafana Dashboards are not working well without container labels.

Reply:
I had already removed this part of the change.